extensions/nanoarrow_ipc/CMakeLists.txt

```diff
@@ -199,8 +216,20 @@ if (NANOARROW_IPC_BUILD_TESTS)
     target_link_libraries(nanoarrow_ipc PRIVATE ipc_coverage_config)
   endif()
 
-  target_link_libraries(nanoarrow_ipc_decoder_test nanoarrow_ipc nanoarrow arrow_shared gtest_main)
-  target_link_libraries(nanoarrow_ipc_reader_test nanoarrow_ipc nanoarrow gtest_main)
+  # On Windows, dynamically linking Arrow is difficult to get right,
+  # so use static linking by default.
+  if (MSVC)
+      set(NANOARROW_IPC_ARROW_TARGET arrow_static)
+  else()
+      set(NANOARROW_IPC_ARROW_TARGET arrow_shared)
+  endif()
```

Comment:
This could also be done with a generator expression something like (untested) `set(NANOARROW_IPC_ARROW_TARGET $<$<${MSVC}>:arrow_static>$<$<NOT:$<${MSVC}>>:arrow_shared>)` but I am not really convinced that it is more readable :D